server/pr_comment_handler.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v32/github"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/pkg/errors"
+)
+
+type prCommentEvent struct {
+	Action     string                     `json:"action"`
+	Comment    *github.PullRequestComment `json:"comment"`
+	Issue      *github.Issue              `json:"issue"`
+	Repository *github.Repository         `json:"repository"`
+}
+
+func (s *Server) prCommentEventHandler(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultRequestTimeout*time.Second)
+	defer cancel()
+
+	ev, err := prCommentEventFromJSON(r.Body)
+	if err != nil {
+		mlog.Error("could not parse pr comment event", mlog.Err(err))
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	defer r.Body.Close()
+
+	pr, err := s.getPRFromEvent(ctx, ev)
+	if err != nil {
+		mlog.Error("Error getting PR from Comment", mlog.Err(err))
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	var commenter string
+	if ev.Comment != nil {
+		commenter = ev.Comment.GetUser().GetLogin()
```

Comment:
GetUser() could return nil as well so we should include that check in the conditional

Reply:
Solved above